apiserver/crossmodel/crossmodel.go

```diff
@@ -53,37 +62,77 @@ func (api *API) Offer(all params.CrossModelOffers) (params.ErrorResults, error)
 	// export service offers
 	offers := make([]params.ErrorResult, len(all.Offers))
 	for i, one := range all.Offers {
-		offer, err := ParseOffer(one)
+		offer, err := parseOffer(one)
 		if err != nil {
 			offers[i].Error = common.ServerError(err)
 			continue
 		}
 
-		if err := crossmodel.ExportOffer(offer); err != nil {
+		if err := api.exporter.ExportOffer(offer); err != nil {
 			offers[i].Error = common.ServerError(err)
 		}
 	}
 	return params.ErrorResults{Results: offers}, nil
 }
 
-func ParseOffer(p params.CrossModelOffer) (crossmodel.Offer, error) {
+// Show gets details about remote services that match given URLs.
+func (api *API) Show(urls []string) (params.RemoteServiceInfoResults, error) {
+	found, err := api.exporter.Search(urls)
```

Comment:
please take some time to look at other bulk queries we have in the codebase, and understand how they should work

the overall method must not fail just because there were no URLs found. the *individual* results should have NotFound errors if the individual URL could not be found. even if none are found, the method should return a set of results with all NotFound errors but, the overall method should not return an error in that case. ONLY if the search fails should you return an error.

also, if a bad URL will cause Search to fail, you cannot make a single Search call. the URLs should be treated independently, with valid results returned for valid URLs, and error results returned for invalid URLs.

assuming Search will only fail if e.g. a network failure occurs, you would rewrite this method as:
 1. search for URLs, and ONLY return `(Results{}, err)` if the search returns an error
 2. `results := make([]Result, len(urls))` (NOT len(found))
 3. for each url, set the result in the same position. if the search returned no info for that url, set a NotFound error.